hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsContext.java

```diff
@@ -142,7 +142,7 @@ public void notifyListeners(Set<DefinitionsEvent> notifications) {
                     log.debugf("Notified Listener %s of %s", e.getKey(), notificationTypes);
                     e.getKey().onChange(notifications.stream()
                             .filter(de -> e.getValue().contains(de.getType()))
-                            .collect(Collectors.toSet()));
+                            .collect(Collectors.toCollection(LinkedHashSet::new)));
```

Comment:
Yes. I think it makes sense to move this signature to List instead Set.
Checking if there is some side effect on this.
I will update the commit.